File: fixtures/scale/elixir/large/lib/ciphers/aria192/impl.ex
defmodule Ciphers.Aria192.Impl do
  alias Ciphers.Aria192.Model

  def encrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 + key + Model.rounds() + Model.key_bits(), 256))
  end

  def decrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 - key - Model.rounds() - Model.key_bits() + 512, 256))
  end
end

File: fixtures/scale/haskell/large/Ciphers/Magma256/Impl.hs
module Ciphers.Magma256.Impl (encrypt, decrypt) where

import Ciphers.Magma256.Model (keyBits, rounds)

encrypt :: Int -> [Int] -> [Int]
encrypt key = map (\b -> mod (b + key + rounds + keyBits) 256)

decrypt :: Int -> [Int] -> [Int]
decrypt key = map (\b -> mod (b - key - rounds - keyBits + 512) 256)
